src/Sarif.Multitool.Library/Rules/SARIF2016.FileUrisShouldBeRelative.cs

```diff
@@ -49,6 +49,40 @@ protected override void Analyze(Run run, string runPointer)
             }
         }
 
+        protected override void Analyze(ArtifactLocation artifactLocation, string artifactLocationPointer)
+        {
+            if (artifactLocation.Uri == null)
+            {
+                return;
+            }
+
+            string uri = artifactLocation.Uri.ToString();
+            if (artifactLocation.Uri.IsAbsoluteUri)
+            {
+                if (artifactLocation.Uri.IsFile)
+                {
+                    if (uri.EndsWith("/"))
+                    {
+                        // TODO: we have a problem, since files can't end with slash
+                    }
+                }
+                else
+                {
+                    if (!uri.EndsWith("/"))
+                    {
+                        // TODO: we have a problem, since folders should end with slash
+                    }
+                }
+            }
+            else
+            {
+                if (uri.StartsWith("/"))
+                {
+                    // TODO: we have a problem, since relative paths should not start with slash
```

Comment:
> // TODO: we have a problem, since relative paths should not start with slash [](start = 19, length = 77)

Yes, for this case, emit an error message (you have a relative url that is prefixed with a slash.